app/workers/notifications/mention_worker.rb

```diff
@@ -0,0 +1,11 @@
+module Notifications
+  class MentionWorker
+    include Sidekiq::Worker
+    sidekiq_options queue: :high_priority, retry: 10
+
+    def perform(mention_id)
+      mention = Mention.find_by(id: mention_id)
+      Notifications::NewMention::Send.call(mention) if mention
```

Comment:
This used to be passed in the old job code as an object. I am using it here directly, let me know you would like a further refactor

Reply:
I think it's fine in this instance, because what you call is a class method, so there's no risk of mocking the wrong instance!

Reply:
@rhymes separate from this PR, we have lots of jobs that do `send if object` which means when we dont have one we silently fail. What do you think about that? Do you think we should change that pattern to raise an error possibly?

Reply:
I think @lightalloy has a better point of view on this but in general IIRC the gist is that we don't need to be alerted for resources that are not present in the DB 99% of the time. The other 1% we need to figure out case by case.

I think we shouldn't take this decision while we're doing this "moving" sprint. Let's move over the jobs to Sidekiq and then we can go through the jobs and decide which ones are worth bubbling the errors for.

Reply:
@rhymes is right about the motivation of adding the check. The job doesn't actually fail as it used to do before adding this checks, but gets canceled.